stress/shell/src/main/java/alluxio/stress/cli/client/StressClientIO.java

```diff
@@ -0,0 +1,377 @@
+/*
+ * The Alluxio Open Foundation licenses this work under the Apache License, version 2.0
+ * (the "License"). You may not use this work except in compliance with the License, which is
+ * available at www.apache.org/licenses/LICENSE-2.0
+ *
+ * This software is distributed on an "AS IS" basis, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND,
+ * either express or implied, as more fully set forth in the License.
+ *
+ * See the NOTICE file distributed with this work for information regarding copyright ownership.
+ */
+
+package alluxio.stress.cli.client;
+
+import alluxio.conf.PropertyKey;
+import alluxio.stress.BaseParameters;
+import alluxio.stress.cli.Benchmark;
+import alluxio.stress.client.ClientIOParameters;
+import alluxio.stress.client.ClientIOTaskResult;
+import alluxio.stress.client.Operation;
+import alluxio.util.CommonUtils;
+import alluxio.util.FormatUtils;
+import alluxio.util.executor.ExecutorServiceFactories;
+
+import com.beust.jcommander.ParametersDelegate;
+import org.apache.hadoop.conf.Configuration;
+import org.apache.hadoop.fs.FSDataInputStream;
+import org.apache.hadoop.fs.FSDataOutputStream;
+import org.apache.hadoop.fs.FileSystem;
+import org.apache.hadoop.fs.Path;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.IOException;
+import java.net.URI;
+import java.nio.ByteBuffer;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Comparator;
+import java.util.List;
+import java.util.Map;
+import java.util.Random;
+import java.util.concurrent.Callable;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.TimeUnit;
+
+/**
+ * Single node client IO stress test.
+ */
+public class StressClientIO extends Benchmark<ClientIOTaskResult> {
+  private static final Logger LOG = LoggerFactory.getLogger(StressClientIO.class);
+
+  @ParametersDelegate
+  private ClientIOParameters mParameters = new ClientIOParameters();
+
+  private FileSystem[] mCachedFs;
+
+  /**
+   * Creates instance.
+   */
+  public StressClientIO() {
+  }
+
+  /**
+   * @param args command-line arguments
+   */
+  public static void main(String[] args) {
+    mainInternal(args, new StressClientIO());
+  }
+
+  @Override
+  public void prepare() throws Exception {
+    if (mBaseParameters.mCluster) {
+      throw new IllegalArgumentException(this.getClass().getName()
+          + " is a single-node client IO stress test, so it cannot be run in cluster mode.");
+    }
+    if (FormatUtils.parseSpaceSize(mParameters.mFileSize) < FormatUtils
+        .parseSpaceSize(mParameters.mBufferSize)) {
+      throw new IllegalArgumentException(String
+          .format("File size (%s) must be larger than buffer size (%s)", mParameters.mFileSize,
+              mParameters.mBufferSize));
+    }
+    if (mParameters.mOperation == Operation.Write) {
+      // Cannot write repeatedly, so warmup is not possible
```

Comment:
Would really like these "warning messages" to be displayed to the user.

Reply:
I cannot print it to the output, but I added a log message for it.